cpp/log/cert.cc

```diff
@@ -233,8 +262,8 @@ string Cert::PrintTime(ASN1_TIME* when) {
 }
 
 
-Cert::Status Cert::IsIdenticalTo(const Cert& other) const {
-  return X509_cmp(x509_, other.x509_) == 0 ? TRUE : FALSE;
+StatusOr<bool> Cert::IsIdenticalTo(const Cert& other) const {
+  return StatusOr<bool>(X509_cmp(x509_, other.x509_) == 0 ? true : false);
```

Comment:
Yes, I'm thinking specifically about the cert.cc code where some operations
call several other methods that could return a mix of StatusOr types it
would be easier to make an error in what then gets returned. I don't know
what's best yet and am still thinking it through.

On 28 August 2015 at 13:29, Pierre Phaneuf <notifications@github.com> wrote:

> In cpp/log/cert.cc
> <https://github.com/google/certificate-transparency/pull/885#discussion_r38194747>
> :
>
> > @@ -233,8 +262,8 @@ string Cert::PrintTime(ASN1_TIME* when) {
> >  }
> >
> >
> > -Cert::Status Cert::IsIdenticalTo(const Cert& other) const {
> > -  return X509_cmp(x509_, other.x509_) == 0 ? TRUE : FALSE;
> > +StatusOr<bool> Cert::IsIdenticalTo(const Cert& other) const {
> > +  return StatusOr<bool>(X509_cmp(x509_, other.x509_) == 0 ? true : false);
>
> It's nothing specific to StatusOr, though, if you have a function that
> returns bool, you can return 42; just fine (that's "true"), and we can't
> really say that we can avoid int returns. [image: :wink:]
>
> In other words, returning StatusOr<Foo> behaves pretty closely to how it
> would if you had defined the function as returning Foo, whatever that may
> be. And integer types in C++ convert rather merrily between each others
> (lest too much C compatibility was lost), so that's something we have to
> live with.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/885/files#r38194747>
> .
>